### PR TITLE
Akeneo styleguides

### PR DESCRIPTION
**Pull request by pierallard**

### CODE REVIEW

File: styleguide/config.yml [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+        Square: { class: 'AknActionButton--square', text: '>' }
+    AknSeveralActionsButton:
+      description: "Prints a button with several actions: a primary action and secondary actions. The text of the
+      primary action is displayed on the button, and a caret is available to display the other actions."
+      modifiers:
+        Apply button: { class: 'AknSeveralActionsButton--apply' }
+        Action button: { class: 'AknSeveralActionsButton--action' }
+        Small button: { class: 'AknSeveralActionsButton--small' }
+    AknDropdownButton:
+      description: "Prints a button to choose the action on a dropdown (there is no primary action)."
```
Comment (rybus) [inline styleguide/config.yml]: on a > within

Comment (rybus) [inline styleguide/config.yml]: within a

File: styleguide/config.yml [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+
+  Templates:
+    AknGridToolbar: ~
+    AknAttributeActions:
+      modifiers:
+        Comparison mode: { template: 'comparisonMode' }
+    AknBreadcrumbContainer: ~
+    AknColumnConfigurator: ~
+    AknFooter:
+      description: "Prints a discreet list of informations."
```
Comment (rybus) [inline styleguide/config.yml]: information is uncountable, remove its "s" at the end

File: styleguide/config.yml [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+    AknHeader: ~
+    AknMainMenu: ~
+    AknNotification:
+      description: "TODO Add modifiers and check why the icon is not green anymore"
```
Comment (rybus) [inline styleguide/config.yml]: Is the "TODO" relevant here?

File: styleguide/config.yml [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+  Panels:
+    AknPanelContainer:
+      description: "Prints a panel for lateral information. The PanelContainer has an height of 100% of the page."
+    AknCommentPanel: ~
+    AknCompletenessBlock: ~
+    AknTabContainer: ~
+
+  Forms:
+    AknFormContainer:
+      description: "Prints a container with a fixed width for listing form elements vertically."
```
Comment (rybus) [inline styleguide/config.yml]: Prints a fixed-width container to list form elements vertically.

File: styleguide/config.yml [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+      modifiers:
+        With padding: { class: 'AknFormContainer--withPadding' }
+    AknFieldContainer:
+      description: "A Field container is mainly composed of a label and an input. You can create a very simple field
+      container with these 2 elements, or add a footer (for validation errors, for example), add icons next to the
```
Comment (rybus) [inline styleguide/config.yml]: thse 2 elements. You could also add a footer [...]

File: styleguide/config.yml [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+      input, etc. Basically, a field container will take all the width of the parent container. If you want resize it,
```
Comment (rybus) [inline styleguide/config.yml]: all the width available from the parent container.

Comment (rybus) [inline styleguide/config.yml]: You will have to use the AknFormContainer module if you want to resize it.

File: styleguide/config.yml [hunk 7]
```diff
@@ -0,0 +1,211 @@
...
+        Without left radius: { class: 'AknTextField--noLeftRadius' }
+        Without radius: { class: 'AknTextField--noRadius' }
+        With right button: { template: 'withRightButton' }
+        Disabled: { class: 'AknTextField--disabled' }
+    AknTextareaField:
+      modifiers:
+        Small: { class: 'AknTextareaField--small' }
+        Disabled: { attribute: 'disabled' }
+    AknMetricField:
+      description: "Prints an inline field with a text field and a dropdown menu to choose metric."
```
Comment (rybus) [inline styleguide/config.yml]: with > containing

Comment (rybus) [inline styleguide/config.yml]: "metric" > "unit"? I am not sure about the context of this button

File: styleguide/config.yml [hunk 8]
```diff
@@ -0,0 +1,211 @@
...
+
+  Navigation:
+    AknHorizontalNavtab: ~
+    AknTabHeader: ~
+    AknVerticalNavtab:
+      modifiers:
+        Condensed: { class: 'AknVerticalNavtab--condensed' }
+    AknBreadcrumb: ~
+    AknUserMenu:
+      description: "TODO: Remove this module and put into Header"
```
Comment (rybus) [inline styleguide/config.yml]: and place it inside Header

File: styleguide/config.yml [hunk 9]
```diff
@@ -0,0 +1,211 @@
...
+      modifiers:
+        Big: { class: 'AknCaret--big' }
+        Inverse: { class: 'AknCaret--inverse' }
+    AknDropdown: ~
+    AknFlash:
+      modifiers:
+        Success: { class: 'AknFlash--success' }
+        Danger: { class: 'AknFlash--danger' }
+    AknLoadingMask:
+      description: "The loading mask is used when an element is loading. Please note than all the style attributes in
```
Comment (rybus) [inline styleguide/config.yml]: or you can also say "is being loaded", both are okay

File: styleguide/config.yml [hunk 10]
```diff
@@ -0,0 +1,211 @@
...
+        Without border: { class: 'AknVerticalList--withoutBorder' }
+        Selectable items: { itemClass: 'selectableItems' }
+        Movable items: { template: 'movableItems' }
+    AknAcl:
+      modifiers:
+        Granted: { iconClass: 'AknAcl-icon--granted', icon: 'icon-ok' }
+        Non granted: { iconClass: 'AknAcl-icon--nonGranted', icon: 'icon-remove' }
+        Centered: { class: 'AknAcl--centered' }
+    AknDiff:
+      description: "Allow to display text with differences (add or remove). Basically, the style only display inline
```
Comment (rybus) [inline styleguide/config.yml]: displays

Comment (rybus) [inline styleguide/config.yml]: so you have to use [...]

File: styleguide/templates/overview.html.twig [hunk 11]
```diff
@@ -0,0 +1,87 @@
+<div class="AknStyleGuide-element" data-group="Overview">
+    <h1 class="AknStyleGuide-elementTitle">
+        Overview
+    </h1>
+
+    <p>
+        In these pages you will find the complete list of available modules for Akeneo PIM. Akeneo PIM use the
```
Comment (rybus) [inline styleguide/templates/overview.html.twig]: uses

File: styleguide/templates/overview.html.twig [hunk 12]
```diff
@@ -0,0 +1,87 @@
...
+        <a href="http://getbem.com/introduction/">BEM methodology</a> to manage the style of the page.
+    </p>
+    <p>
+        BEM means:
+        <ul>
+            <li><strong>Block</strong>: an designed entity,</li>
```
Comment (rybus) [inline styleguide/templates/overview.html.twig]: a designed

File: styleguide/templates/overview.html.twig [hunk 13]
```diff
@@ -0,0 +1,87 @@
...
+            <li><strong>Element</strong>: a part of the block,</li>
+            <li><strong>Modifier</strong>: a variation on a block or on an element.</li>
+        </ul>
+    </p>
+    <p>
+        Different naming conventions are available for BEM methodology. In Akeneo PIM, we implement the next convention:
```
Comment (rybus) [inline styleguide/templates/overview.html.twig]: for the

File: styleguide/templates/overview.html.twig [hunk 14]
```diff
@@ -0,0 +1,87 @@
...
+        <ul>
+            <li><tt class="docutils literal"><span class="pre">AknBlockName</span></tt> for the main block declaration.
+                We use the prefix Akn to distinguish the Akeneo blocks from external libraries. The block name is in
```
Comment (rybus) [inline styleguide/templates/overview.html.twig]: to distinguish Akeneo blocks from those of external libraries

File: styleguide/templates/overview.html.twig [hunk 15]
```diff
@@ -0,0 +1,87 @@
...
+                upper CamelCase.</li>
+            <li><tt class="docutils literal"><span class="pre">AknBlockName-elementName</span></tt> for each element of
+                the block. The element is linked to the block name with a simple dash, and use lower camelCase.</li>
+            <li><tt class="docutils literal"><span class="pre">AknBlockName-elementName--modifierName</span></tt> or
+                <tt class="docutils literal"><span class="pre">AknBlockName--modifierName</span></tt> for the modifiers.
+                The modifier is linked to the block or element with 2 dashes and use lower camelCase</li>
```
Comment (rybus) [inline styleguide/templates/overview.html.twig]: and uses

File: styleguide/templates/overview.html.twig [hunk 16]
```diff
@@ -0,0 +1,87 @@
...
+        </ul>
+    </p>
+
+    <hr/>
+
+    <p>
+        You can use every block in your custom development, simply with a copy-paste of the HTML code.
```
Comment (rybus) [inline styleguide/templates/overview.html.twig]: Not sure of the relevancy of this sentence, to me it is implicit

File: styleguide/templates/overview.html.twig [hunk 17]
```diff
@@ -0,0 +1,87 @@
...
+    <pre>
+    <code class="html">&#60;div class="AknButtonList AknButtonList--centered"&#62;
+    &#60;div class="AknButtonList-item">Item 1&#60;/div&#62;
+    &#60;div class="AknButtonList-item">Item 2&#60;/div&#62;
+&#60;/div&#62;</code>
+    </pre>
+
+    <p>
+        The <tt class="docutils literal"><span class="pre">AknButtonList-item</span></tt> is an element of
+        <tt class="docutils literal"><span class="pre">AknButtonList</span></tt>, but can be a module itself, for example
```
Comment (rybus) [inline styleguide/templates/overview.html.twig]: but can also be

File: styleguide/README.md [hunk 18]
```diff
@@ -0,0 +1,41 @@
+Akeneo style guide
+==================
+
+The style guides are automatically generated in a single page index.php.
+When you update a BEM module in the pim-community-dev repository, you have to update the style guide.
+
+How it works
+------------
+
+- The file `styleguide/config.yml` contains the list of the available module, their description and modifiers.
```
Comment (SamirBoulil) [inline styleguide/README.md]: available modules. their descriptions and modifiers

File: styleguide/README.md [hunk 19]
```diff
@@ -0,0 +1,41 @@
...
+- Each module has its own folder in `templates/`. The default template is called `base.html.twig`.
+- You can simply access to `styleguide/index.php` to view the style guides.
```
Comment (MarieMinasyan) [inline styleguide/README.md]: access (no _to_)

File: styleguide/templates/overview.html.twig [hunk 20]
```diff
@@ -0,0 +1,88 @@
...
+    <p>
+        BEM means:
+        <ul>
+            <li><strong>Block</strong>: a designed entity,</li>
+            <li><strong>Element</strong>: a part of the block,</li>
+            <li><strong>Modifier</strong>: a variation on a block or on an element.</li>
+        </ul>
+    </p>
+    <p>
+        Different naming conventions are available for the BEM methodology. In Akeneo PIM, we implement the next
```
Comment (MarieMinasyan) [inline styleguide/templates/overview.html.twig]: the following convention

Comment (fred2gombert) [inline styleguide/templates/overview.html.twig]: the following convention (instead of next)

File: styleguide/config.yml [hunk 21]
```diff
@@ -0,0 +1,206 @@
...
+        Without border: { class: 'AknVerticalList--withoutBorder' }
+        Selectable items: { itemClass: 'selectableItems' }
+        Movable items: { template: 'movableItems' }
+    AknAcl:
+      modifiers:
+        Granted: { iconClass: 'AknAcl-icon--granted', icon: 'icon-ok' }
+        Non granted: { iconClass: 'AknAcl-icon--nonGranted', icon: 'icon-remove' }
+        Centered: { class: 'AknAcl--centered' }
+    AknDiff:
+      description: "Displays text with differences (add or remove). Basically, the style only display inline text, so you have to use one of the available modifiers to see the difference."
```
Comment (MarieMinasyan) [inline styleguide/config.yml]: displays

File: styleguide/config.yml [hunk 22]
```diff
@@ -0,0 +1,206 @@
...
+        Big: { class: 'AknCaret--big' }
+        Important: { class: 'AknCaret--important' }
+        Apply: { class: 'AknCaret--apply' }
+        Inverse: { class: 'AknCaret--inverse', dark: true }
+    AknFlash:
+      modifiers:
+        Success: { class: 'AknFlash--success' }
+        Danger: { class: 'AknFlash--danger' }
+    AknLoadingMask:
+      description: "The loading mask is used when an element is being loaded. Please note than all the style attributes in the source example are automatically generated by the application, you don't need to add it."
```
Comment (MarieMinasyan) [inline styleguide/config.yml]: _them_ instead of _it_

File: styleguide/config.yml [hunk 23]
```diff
@@ -0,0 +1,206 @@
...
+    AknTitleContainer: ~
+    AknWidget:
+      modifiers:
+        withPadding: { contentClass: 'AknWidget-content--withPadding' }
+    AknInfoBlock:
+      description: "Prints a centered text for additional information."
+
+  Panels:
+    AknPanelContainer:
+      description: "Prints a panel for lateral information. The PanelContainer has an height of 100% of the page."
```
Comment (MarieMinasyan) [inline styleguide/config.yml]: _a height_

File: styleguide/templates/overview.html.twig [hunk 24]
```diff
@@ -0,0 +1,88 @@
+<div class="AknStyleGuide-element" data-group="Overview">
+    <h1 class="AknStyleGuide-elementTitle">
+        Overview
+    </h1>
+
+    <p>
+        In these pages you will find the complete list of available modules for Akeneo PIM. Akeneo PIM uses the
```
Comment (fred2gombert) [inline styleguide/templates/overview.html.twig]: In the following pages you will find a comprehensive list of available modules for Akeneo PIM. We organize our styles using the BEM methodology

File: styleguide/config.yml [hunk 25]
```diff
@@ -0,0 +1,206 @@
...
+        Disabled: { attribute: 'disabled' }
+    AknMetricField:
+      description: "Prints an inline field containing a text field and a dropdown menu to choose unit."
+    AknPriceList:
+      description: "Prints a list of prices. The width of each price is computed to display 3 inline prices on an AknFormContainer"
+    AknMediaField:
+      modifiers:
+        With media: { template: 'withMedia' }
+    AknComparableFields:
+      description: "Displays 2 FieldContainer side by side, with a checkbox between."
```
Comment (MarieMinasyan) [inline styleguide/config.yml]: in between

File: styleguide/config.yml [hunk 26]
```diff
@@ -0,0 +1,206 @@
...
+        Without left radius: { class: 'AknTextField--noLeftRadius' }
+        Without radius: { class: 'AknTextField--noRadius' }
+        With right button: { template: 'withRightButton' }
+        Disabled: { class: 'AknTextField--disabled' }
+    AknTextareaField:
+      modifiers:
+        Small: { class: 'AknTextareaField--small' }
+        Disabled: { attribute: 'disabled' }
+    AknMetricField:
+      description: "Prints an inline field containing a text field and a dropdown menu to choose unit."
```
Comment (MarieMinasyan) [inline styleguide/config.yml]: a unit

File: styleguide/config.yml [hunk 27]
```diff
@@ -0,0 +1,206 @@
...
+      description: "Prints a set of grids in one line."
+    AknGridContainer: ~
+
+  Forms:
+    AknFormContainer:
+      description: "Prints a fixed-width container for listing form elements vertically."
+      modifiers:
+        With padding: { class: 'AknFormContainer--withPadding' }
+    AknFieldContainer:
+      description: "A Field container is mainly composed of a label and an input. You can create a very simple field container with these 2 elements. You can also add a footer (for validation errors, for example), add icons next to the input, etc. Basically, a field container will take all the width of the parent container. If you want resize it, use the AknFormContainer module."
```
Comment (MarieMinasyan) [inline styleguide/config.yml]: to resize

File: styleguide/config.yml [hunk 28]
```diff
@@ -0,0 +1,206 @@
+groups:
+  Buttons:
+    AknButton:
+      description: "Prints buttons with text, with or without an icon. You can mix this module's modifiers: for example, you can create a big important uppercase button with icon. If the button contains only an icon, please use the AknIconButton module."
```
Comment (MarieMinasyan) [inline styleguide/config.yml]: uppercase button with _an_ icon

**Pull request merged**